### PR TITLE
Warn on max_header_lines

### PR DESCRIPTION
**Pull request by KingNoosh**
Fixes #

Changes proposed in this pull request:
- 
- 

Checklist:
- [ ] docs updated
- [ ] tests updated
- [ ] [Changes](https://github.com/haraka/Haraka/blob/master/Changes.md) updated

### CODE REVIEW

**Comment by msimerson** [comment]
We've already got `const MAX_HEADER_LINES = config.get('max_header_lines') || 1000;` in `./transaction.js.` Does it make sense to export MAX_HEADER_LINES so you can reference it from `connection.js`?

**Comment by KingNoosh** [comment]
I can take a further look.

File: connection.js [hunk 1]
```diff
@@ -1583,6 +1583,12 @@ class Connection {
             return;
         }
 
+        // Warn if we hit the maximum parsed header lines limit
+        const max_lines = trans.MAX_HEADER_LINES;
+        if (this.transaction.header_lines.length >= max_lines) {
```
Comment (msimerson) [inline connection.js]: What's the utility of creating max_lines?  You can now:

`if (this.transaction.header_lines.length >= trans.MAX_HEADER_LINES) {`

right?

Comment (KingNoosh) [reply connection.js]: Makes sense since we're not using it more than once

**Pull request merged**